Install from source instructions should list golang-13+ as required version.

### Describe the feature or problem you’d like to solve
Install from source instructions 
https://github.com/cli/cli/blob/master/source.md
should list golang-13+ as required version.

### Proposed solution
Document the requirement and have the Makefile check the available version of go before doing anything else to avoid wasting download bandwidth and people's time.

### Additional context
On Ubuntu-19.10, I get:
...
go: downloading github.com/dlclark/regexp2 v1.1.6
go: downloading github.com/danwakefield/fnmatch v0.0.0-20160403171240-cbb64ac3d964
go: extracting github.com/danwakefield/fnmatch v0.0.0-20160403171240-cbb64ac3d964
go: extracting golang.org/x/crypto v0.0.0-20190530122614-20be4c3c3ed5
go: extracting github.com/dlclark/regexp2 v1.1.6
go: extracting gopkg.in/yaml.v3 v3.0.0-20191010095647-fc94e3f71652
go: extracting golang.org/x/sys v0.0.0-20190813064441-fde4db37ae7a
# github.com/cli/cli/context
context/config_file.go:22:19: undefined: errors.Is
note: module requires Go 1.13
make: *** [Makefile:13: bin/gh] Error 2

--- a/source.md
+++ b/source.md
@@ -1,5 +1,11 @@
 # Installation from source
 
+0. Verify that you have Go 1.13+ installed
+```
+$ go version
+go version go1.13.7
+```
+
 1. Clone cli into `~/.githubcli`
 ```
 $ git clone https://github.com/cli/cli.git ~/.githubcli
@@ -9,8 +15,6 @@ $ git clone https://github.com/cli/cli.git ~/.githubcli
 ```
 $ cd ~/.githubcli && make
 ```
-
-_Note: it requires Go 1.13+_ 
 
 3. Add `~/.githubcli/bin` to your $PATH for access to the gh command-line utility.
 

--- a/source.md
+++ b/source.md
@@ -10,6 +10,8 @@ $ git clone https://github.com/cli/cli.git ~/.githubcli
 $ cd ~/.githubcli && make
 ```
 
+_Note: it requires Go 1.13+_ 
+
 3. Add `~/.githubcli/bin` to your $PATH for access to the gh command-line utility.
 
   * For **bash**:

